### PR TITLE
Feature/improve yaw calibration

### PR DESCRIPTION
**Pull request by simenbootsma**
Improves the yaw calibration by comparing the xsens yaw from 50 ms ago to the freshly received vision yaw. Also updates the vision yaw with the rate of turn data if no new vision is received. Tested it, works great. Watchdog is already included here.

### CODE REVIEW

File: Lib/yawCalibration.c [hunk 1]
```diff
@@ -13,37 +15,44 @@ static bool hasCalibratedOnce = false;
...
 	static float yawOffset = 0.0f;
 	static int restCounter = 0;
 	static float sumXsensVec[2] = {0.0f};
 	static float sumVisionVec[2] = {0.0f};
 	static float prevVisionYaw = 0.0f;
 
-	//if (visionYaw == prevVisionYaw) {
-	//	visionYaw += MT_GetGyro()[2] * TIME_DIFF;
-	//}
+	if (visionYaw == prevVisionYaw) {
```
Comment (simenbootsma) [inline Lib/yawCalibration.c]: Please let me know what you think of this part.

Comment (casdoorn) [reply Lib/yawCalibration.c]: visionYaw is not updated, it's a local copy in this function. So changes will only aply to the local copy, which is discarded after function exit

Comment (casdoorn) [reply Lib/yawCalibration.c]: maybe make it explicit that it is not the received vision yaw but the interpolated version

File: Lib/yawCalibration.c [hunk 2]
```diff
@@ -57,7 +66,21 @@ void yaw_Calibrate(float xsensYaw, float visionYaw, bool visionAvailable) {
...
+//	if (HAL_GetTick() - printtime > 20) {
+//		n += sprintf(msg + n, "vision: %f\n\r", visionYaw);
+//		n += sprintf(msg + n, "new xsens: %f\n\r", newXsensYaw);
+//		n += sprintf(msg + n, "old xsens: %f\n\r", oldXsensYaw);
+//		n += sprintf(msg + n, "calibrated: %f\n\r", calibratedYaw);
+//		n += sprintf(msg + n, "offset: %f\n\r", yawOffset);
+//		n += sprintf(msg + n, "bool: %u\n\r", calibratedThisTick);
+//		Putty_printf(msg);
+//		printtime = HAL_GetTick();
+//	}
```
Comment (kJh19) [inline Lib/yawCalibration.c:79]: So whats this?

Comment (simenbootsma) [reply Lib/yawCalibration.c:79]: Prints for debugging purposes

**Review by kJh19 — APPROVED** [review]

**Pull request merged**